### PR TITLE
multi-architecture docker image

### PR DESCRIPTION
**Pull request by anshal-savla**
Ticket: https://nearinc.atlassian.net/browse/DPI-65

updated bash script to build multi-architecture Docker images (arm65/amd64) for contract-builder.

### CODE REVIEW

File: contract-builder/build.sh [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+    docker buildx use --builder contract-builder
+else
+    docker buildx create --name contract-builder --use 
+fi
+
+if [[ ${branch} == "master" ]];then
+    docker buildx build --platform linux/amd64,linux/arm64 -t nearprotocol/"${image_name}:${branch}-${commit}" -t nearprotocol/${image_name}:latest --push .
+else 
+    docker buildx build --platform linux/amd64,linux/arm64 -t nearprotocol/"${image_name}:${branch}-${commit}" --push .
+fi
```
Comment (itegulov) [inline contract-builder/build.sh:23]:
```suggestion
if docker buildx ls | grep -q contract-builder; then
    docker buildx use --builder contract-builder
else
    docker buildx create --name contract-builder --use
fi

if [[ ${branch} == "master" ]]; then
    docker buildx build --platform linux/amd64,linux/arm64 -t nearprotocol/"${image_name}:${branch}-${commit}" -t nearprotocol/${image_name}:latest --push .
else
    docker buildx build --platform linux/amd64,linux/arm64 -t nearprotocol/"${image_name}:${branch}-${commit}" --push .
fi
```

**Review by itegulov — COMMENTED** [review]
Seems to work fine on both Linux and macOS with M1, but I am honestly not sure if contract-builder image related scripts and logic belong to this repo. If they do, shouldn't we also have a buildkite workflow that invokes this script? How can I try it out?

**Comment by itegulov** [comment]
Also, while we are at it and before we publish the next version of the image, can we also update Dockerfile to use Rust 1.56 to unblock https://github.com/near/near-sdk-rs/pull/669?

**Review by mhalambek — APPROVED** [review]

**Review by austinabell — APPROVED** [review]
don't really see why this would exist within the SDK, but I suppose this could be moved out later

**Comment by anshal-savla** [comment]
@itegulov @austinabell  Yes, I can update the rust version in the Docker image. I am Creating a buildkite pipeline to build and publish the docker image in the near-ops repo.

Where should this build script live if not in this repo?

**Comment by austinabell** [comment]
> @itegulov @austinabell Yes, I can update the rust version in the Docker image. I am Creating a buildkite pipeline to build and publish the docker image in the near-ops repo.
> 
> Where should this build script live if not in this repo?

I suppose it's fine to be here, but maybe more ideal to move into its own repo in the future. Nothing here is tied to this SDK specifically.

**Comment by austinabell** [comment]
> I suppose it's fine to be here, but maybe more ideal to move into its own repo in the future. Nothing here is tied to this SDK specifically.

Actually, second thought might be beneficial to be within SDK but only if we have building/using the image as part of the CI checks, which doesn't require building for the local script. This could be a benefit to make sure the script doesn't break examples and general use things within the SDK directly.

Otherwise, not a benefit since we just use the deployed image [here](https://github.com/near/near-sdk-rs/blob/master/examples/build_docker.sh).

**Pull request merged**